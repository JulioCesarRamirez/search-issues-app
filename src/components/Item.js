import React from 'react';
import PropTypes from 'prop-types';

export const Item = ({ title, url }) => {
  return (
    <div className="card">
      <a href={url} target="_blank">
        {title}
      </a>
    </div>
  );
};

Item.prototype = {
  issue: PropTypes.string.isRequired,
};
