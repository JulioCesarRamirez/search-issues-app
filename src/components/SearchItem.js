import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const SearchItem = ({ setItemToSearch }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setItemToSearch((items) => [inputValue, ...items]);
      setInputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange}/>
    </form>
  );
};

SearchItem.propTypes = {
  setItemToSearch: PropTypes.func.isRequired,
};
