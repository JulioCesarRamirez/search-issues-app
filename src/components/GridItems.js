import React from 'react';
import PropTypes from "prop-types";

import { useFetch } from '../hooks/useFetch';
import { Item } from './Item';

export const GridItems = ({ item }) => {
  const { data: itemsFromApi } = useFetch(item);
  return (
    <div className="card-grid">
      {itemsFromApi.map((item, i) => (
        <Item key={i} {...item} />
      ))}
    </div>
  );
};

GridItems.protoTypes = {
  item: PropTypes.string.isRequired,
};
