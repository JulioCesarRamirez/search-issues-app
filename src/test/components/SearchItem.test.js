import React from 'react';
import '@testing-library/jest-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import { SearchItem } from '../../components/SearchItem';
configure({ adapter: new Adapter() });

describe('Test in <SearchItem />', () => {
  const setItemToSearch = jest.fn();
  let wrapper = shallow(<SearchItem setItemToSearch={setItemToSearch} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<SearchItem setItemToSearch={setItemToSearch} />);
  });
  it('should show correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should change input value', () => {
    const input = wrapper.find('input');
    const value = 'Hello world';
    input.simulate('change', { target: { value: value } });
  });
  it('should not send information when submit form', () => {
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(setItemToSearch).not.toHaveBeenCalled();
  });
  it('should clear the input and call setItemToSearch', () => {
    const value = 'Hello world';
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: value } });
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(setItemToSearch).toHaveBeenCalled();
    expect(input.prop('value')).toBe('');
  });
});
