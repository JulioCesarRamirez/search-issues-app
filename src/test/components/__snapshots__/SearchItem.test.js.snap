// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchItem /> should show correctly 1`] = `ShallowWrapper {}`;
